docs/developers/contributing.rst

```diff
@@ -39,33 +43,19 @@ For impure functions, i.e. functions with side effects, if you find yourself
 redesigning interfaces or figuring out how to mock something to support unit
 tests, then stop and consider the following strategies instead:
 
-1. Extract the logic you want to test into pure functions.  When done right the
+#. Extract the logic you want to test into pure functions.  When done right the
    side effect would be increased composability, setting you up for future code
    reuse.
-2. Contain the remaining functions with side effects in logical units that
+#. Contain the remaining functions with side effects in logical units that
    can be thoroughly tested in the integration test suite.
 
-### Running tests
-
-#### Prerequisites
-
-* lxd (`sudo snap install lxd`)
-* GNU parallel (`sudo apt -y install parallel`)
-* shellcheck (`sudo apt -y install shellcheck`)
-* snapcraft (`sudo snap install snapcraft`)
-
-#### Lint
-
-Go code is linted as part of the snap build process:
-
-    make
-
-To check lint for the test code:
-
-    make check-lint
+Running tests
+~~~~~~~~~~~~~
 
-#### Functional tests
+..
+  The developer documentation is in the docs/developers directory.
 
-To run the entire suite in serial:
+Please refer to the document on :doc:`testing <testing>` to learn how to :ref:`run MicroOVN tests`.
```

Comment:
Out of curiosity, what's the reason for these two links? They seem to point to pretty much the same location.

Reply:
If you read this from `vim` you need to know which file to look in, and it might as well be a link for anyone reading from the web.